src/pty.rs

```diff
@@ -0,0 +1,128 @@
+//! Create master and slave virtual pseudo-terminals (PTTYs)
+
+use std::ffi::CStr;
+use std::os::unix::io::RawFd;
+
+use libc;
+
+use {Error, fcntl, Result};
+
+/// Grant access to a slave pseudoterminal (see
+/// [grantpt(3)](http://man7.org/linux/man-pages/man3/grantpt.3.html))
+///
+/// `grantpt()` changezs the mode and owner of the slave pseudoterminal device corresponding to the
+/// master pseudoterminal referred to by `fd`. This is a necessary step towards opening the slave.
+#[inline]
+pub fn grantpt(fd: RawFd) -> Result<()> {
```

Comment:
So in continuing to think about this, I don't think it makes sense to provide an abstraction over `RawFd`. The use case of these ttys is generally what you can see in `serialport-rs`. Most commonly you call `read` and `write`, both of which take a `RawFd`. Additionally, the other functions I'm adding here all take `RawFd`. So every call you'd be doing `fn(port.as_raw_fd())`, so it's just a needless abstraction that doesn't actually add any safety. I think it's the role of a library like `serialport-rs` to provide abstractions on top of nix should it be desired.